src/base/bittorrent/session.cpp

```diff
@@ -2134,9 +2165,26 @@ bool Session::addTorrent_impl(CreateTorrentParams params, const MagnetUri &magne
 
     // If empty then Automatic mode, otherwise Manual mode
     QString savePath = params.savePath.isEmpty() ? categorySavePath(params.category) : params.savePath;
+    const bool fromMagnetUri = magnetUri.isValid();
     libt::add_torrent_params p;
+    if (params.restored && !fromMagnetUri) {
+#if LIBTORRENT_VERSION_NUM >= 10200
+        libt::error_code ec;
+        p = libt::read_resume_data(fastresumeData, ec);
+#else
+        // Set torrent fast resume data
+        p.resume_data = {fastresumeData.constData(), fastresumeData.constData() + fastresumeData.size()};
+        p.flags |= libt::add_torrent_params::flag_use_resume_save_path;
+#endif
+    }
+
     InfoHash hash;
-    const bool fromMagnetUri = magnetUri.isValid();
+
+#if LIBTORRENT_VERSION_NUM < 10200
+    std::vector<boost::uint8_t> filePriorities;
+#else
+    std::vector<libt::download_priority_t> filePriorities;
```

Comment:
I suspect you may save conditional code like this by creating a typedef to `download_priority_t`